### PR TITLE
Fix Mac compilation issues

### PR DESCRIPTION
**Pull request by 88hcsif**
Fix the compilation hurdles mentioned in #146 and more.

I guess the change to GLideN64/src/PaletteTexture.h should be pushed upstream first.
However, according to the CI build status, https://github.com/libretro/GLideN64 already compiles on OSX...
I wonder if the travis environment include malloc.h some other way...

### CODE REVIEW

File: GLideN64/src/PaletteTexture.h [hunk 1]
```diff
@@ -1,11 +1,11 @@
 #pragma once
 #include <memory>
 
-#ifdef IOS
+#if defined IOS || defined OSX
```
Comment (m4xw) [inline GLideN64/src/PaletteTexture.h:4]: `#if defined(IOS) || defined(OSX)`

File: libretro-common/glsm/glsm.c [hunk 2]
```diff
@@ -2500,7 +2500,11 @@ void rglProgramBinary(GLuint program,
 
 void rglTexImage2DMultisample( 	GLenum target,
   	GLsizei samples,
+#if defined(OSX)
+  	GLint internalformat,
```
Comment (m4xw) [inline libretro-common/glsm/glsm.c:2504]: I need to double check them, I doubt it differs for IOS so its probably wrong type to begin with

**Comment by estefan3112** [comment]
Hi, the core builds fine in MacOS Mojave, Retroarch loads a version 2.1 2dbff90, but each game crashes immediately. Crash report here?

**Comment by 88hcsif** [comment]
It works fine with the libretro frontend which is part of [cloud-game](https://github.com/giongto35/cloud-game).
Maybe retroarch also does not expect the retro_serialize function to run the run_retro?
But that would also imply that retroarch calls the retro_serialize function without being prompted (which might be true if that is how the rewind functionality works...).
@m4xw is retroarch the frontend you normally use?

**Comment by m4xw** [comment]
I changed it to only do the co_switch earlier today, omitting the video cb

**Comment by m4xw** [comment]
But also if that was the cause, then there is a bug somewhere else in the frontend.

**Comment by 88hcsif** [comment]
Okay, Retroarch has issues even with the code before the rebase, so it should not be related to the retro_serialize changes.
I think it is some issue with how OpenGL is initialised...

**Comment by m4xw** [comment]
oh I guess i killed that PR on rebase by accident.
Can you rebase / reopen this and target develop?

**Pull request closed without merging**